filebeat/crawler/registrar.go

```diff
@@ -74,7 +77,7 @@ func (r *Registrar) LoadState() {
 		defer existing.Close()
 		logp.Info("Loading registrar data from %s", r.registryFile)
 		decoder := json.NewDecoder(existing)
-		decoder.Decode(&r.State)
+		decoder.Decode(r.getFullState())
```

Comment:
@urso That raises the problem, that we actually don't have any tests that check if continue reading a file is done properly after restart.